mParticle-OneTrust/MPKitOneTrust.m

```diff
@@ -94,7 +97,7 @@ -(void)actionConsent:(NSNotification*)notification {
     NSNumber *status = notification.object; // BOOL-ish value for consent
 
     // Fire consent change event
-    [self createConsentEvent:category :status];
+    [self createConsentEvent:category :self.purposeConsentMapping :status];
```

Comment:
It's entirely possible (although OneTrust did say they didn't support it yet) that this listener will also work for vendor consent.  The same listener on web applies to both because it's the same dialogue box that pops up and you have to hit save after making either regular consent, or vendor consent changes.  Might be worth manually testing.